vm/actor/src/builtin/market/state.rs

```diff
@@ -111,60 +362,51 @@ impl State {
         })
     }
 
-    pub(super) fn maybe_lock_balance<BS: BlockStore>(
+    fn maybe_lock_balance<BS: BlockStore>(
         &mut self,
         store: &BS,
         addr: &Address,
         amount: &TokenAmount,
     ) -> Result<(), ActorError> {
         let prev_locked = self.get_locked_balance(store, addr)?;
-        let escrow_balance = self.get_locked_balance(store, addr)?;
+        let escrow_balance = self.get_escrow_balance(store, addr)?;
         if &prev_locked + amount > escrow_balance {
             return Err(ActorError::new(
-                ExitCode::ErrInsufficientFunds,
+                ExitCode::ErrPlaceholder,
                 format!(
-                    "not enough balance to lock for addr {}: {} < {} + {}",
-                    addr, escrow_balance, prev_locked, amount
+                    "not enough balance to lock for addr {}: {} <  {}",
+                    addr,
+                    prev_locked + amount,
+                    escrow_balance
                 ),
             ));
         }
 
-        let mut bt = BalanceTable::from_root(store, &self.locked_table)
-            .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
-        bt.add(addr, amount).map_err(|e| {
-            ActorError::new(
-                ExitCode::ErrIllegalState,
-                format!("adding locked balance {}", e),
-            )
-        })?;
-        self.locked_table = bt
-            .root()
-            .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+        mutate_balance_table(
+            store,
+            &mut self.locked_table,
+            |lt: &mut BalanceTable<BS>| {
+                lt.add(addr, amount)?;
+                Ok(())
+            },
+        )
+        .map_err(|e| ActorError::new(ExitCode::ErrPlaceholder, e))?;
+
         Ok(())
     }
-
-    pub(super) fn unlock_balance<BS: BlockStore>(
+    /// TODO: all these balance table mutations need to happen at the top level and be batched (no flushing after each!)
```

Comment:
I think in general, we don't need to copy their TODOs, we just need to match their functionality so having the TODOs is a bit confusing because it indicates we have something more to do here